src/libraries/System.Net.Mail/src/System/Net/Base64Stream.cs

```diff
@@ -199,104 +199,167 @@ internal int EncodeBytes(byte[] buffer, int offset, int count, bool dontDeferFin
+                    WriteState.LastBits = (byte)((b & 0x0f) << 2);
+                    WriteState.Padding = 1;
+                    return 1;
+                case 1: // Add third byte of 3
+                    WriteState.Append(Base64EncodeMap[WriteState.LastBits | ((b & 0xc0) >> 6)]);
+                    WriteState.Append(Base64EncodeMap[(b & 0x3f)]);
+                    WriteState.Padding = 0;
+                    return 1;
+            }
+
+            return 0;
+        }
+
+        private void AppendPadding()
+        {
+            Debug.Assert(0 <= WriteState.Padding && WriteState.Padding <= 2, "paddind was not in range [0,2]");
+
+            switch (WriteState.Padding)
+            {
+                case 0: // No padding needed
                     break;
+                case 2: // 2 character padding needed (1 byte was encoded instead of 3)
+                    WriteState.Append(Base64EncodeMap[WriteState.LastBits]);
+                    WriteState.Append(Base64EncodeMap[64]);
+                    WriteState.Append(Base64EncodeMap[64]);
+                    WriteState.Padding = 0;
+                    break;
+                case 1: // 1 character padding needed (2 bytes were encoded instead of 3)
+                    WriteState.Append(Base64EncodeMap[WriteState.LastBits]);
+                    WriteState.Append(Base64EncodeMap[64]);
+                    WriteState.Padding = 0;
+                    break;
+            }
+        }
 
-                case 1: // Two character padding needed
-                    WriteState.Append(Base64EncodeMap[(buffer[cur] & 0xFC) >> 2]);
-                    if (dontDeferFinalBytes)
-                    {
-                        WriteState.Append(Base64EncodeMap[(byte)((buffer[cur] & 0x03) << 4)]);
-                        WriteState.Append(Base64EncodeMap[64]);
-                        WriteState.Append(Base64EncodeMap[64]);
-                        WriteState.Padding = 0;
-                    }
-                    else
-                    {
-                        WriteState.LastBits = (byte)((buffer[cur] & 0x03) << 4);
-                        WriteState.Padding = 2;
-                    }
-                    cur++;
+        private bool LineBreakNeeded(int numberOfBytesToAppend)
+        {
+            if (_lineLength == -1)
+            {
+                return false;
+            }
+
+            int bytesLeftInCurrentBlock = 0;
+            int numberOfCharsLeftInCurrentBlock = 0;
+            switch (WriteState.Padding)
+            {
+                case 2: // 1 byte was encoded from 3
+                    bytesLeftInCurrentBlock = 2;
+                    numberOfCharsLeftInCurrentBlock = 3;
+                    break;
+                case 1: // 2 bytes were encoded from 3
+                    bytesLeftInCurrentBlock = 1;
+                    numberOfCharsLeftInCurrentBlock = 2;
                     break;
```

Comment:
`default` case is missing. Is it intentional?

Reply:
Fixed https://github.com/dotnet/runtime/blob/e8c55d73aa809b072b780f38541ba9ccb7e28667/src/libraries/System.Net.Mail/src/System/Net/Mime/Base64Encoder.cs#L76-L80